packages/block-library/src/tag-cloud/edit.js

```diff
@@ -54,6 +78,28 @@ function TagCloudEdit( { attributes, setAttributes, taxonomies } ) {
 		return [ selectOption, ...taxonomyOptions ];
 	};
 
+	const onFontSizeChange = ( fontSize, value ) => {
+		// eslint-disable-next-line @wordpress/no-unused-vars-before-return
+		const [ quantity, newUnit ] = parseUnit( value );
+		if ( ! Number.isFinite( quantity ) || quantity < 0 || quantity > 100 ) {
+			return;
+		}
+		const updateObj = { [ fontSize ]: value };
+		// We need to keep in sync the `unit` changes to both `smallestFontSize`
+		// and `largestFontSize` attributes.
+		Object.entries( {
+			smallestFontSize,
+			largestFontSize,
+		} ).forEach( ( [ attribute, currentValue ] ) => {
+			const [ _value, _unit ] = parseUnit( currentValue );
+			// Only add an update if the other font size attribute has a different unit.
+			if ( attribute !== fontSize && _unit !== newUnit ) {
+				updateObj[ attribute ] = `${ _value }${ newUnit }`;
+			}
+		} );
+		setAttributes( updateObj );
+	};
```

Comment:
Small nit: renaming some of the variables for added clarity

```suggestion
	const onFontSizeChange = ( fontSizeLabel, newValue ) => {
		// eslint-disable-next-line @wordpress/no-unused-vars-before-return
		const [ quantity, newUnit ] = parseUnit( newValue );
		if ( ! Number.isFinite( quantity ) || quantity < 0 || quantity > 100 ) {
			return;
		}
		const updateObj = { [ fontSizeLabel ]: newValue };
		// We need to keep in sync the `unit` changes to both `smallestFontSize`
		// and `largestFontSize` attributes.
		Object.entries( {
			smallestFontSize,
			largestFontSize,
		} ).forEach( ( [ attribute, currentValue ] ) => {
			const [ currentQuantity, currentUnit ] = parseUnit( currentValue );
			// Only add an update if the other font size attribute has a different unit.
			if ( attribute !== fontSizeLabel && currentUnit !== newUnit ) {
				updateObj[ attribute ] = `${ currentQuantity }${ newUnit }`;
			}
		} );
		setAttributes( updateObj );
	};
```